add int16 support

apoorvsxna opened the pull request:
## Pull Request description
- Added support for int16.
- Added test for the same.
- Fixed bug where `named_values` dictionary was being shared across multiple instances.

Fixes #26 and #40 .

## Pull Request checklists

This PR is a:

- [x] bug-fix
- [x] new feature
- [ ] maintenance

About this PR:

- [x] it includes tests.
- [x] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:

- [x] I have reviewed the changes and it contains no misspelling.
- [x] The code is well commented, especially in the parts that contain more
      complexity.
- [x] New and old tests passed locally.

## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .

apoorvsxna commented:
Hey, just went through the contributing guidelines. I'll take care of the commit message format from now on.

xmnlab commented:
Thanks @apoorvsxna for working on that, could you fix the issues in the linting ?

apoorvsxna commented:
Oh alright, let me fix that asap

apoorvsxna commented:
> Thanks @apoorvsxna for working on that, could you fix the issues in the linting ?

Hey @xmnlab . All sorted now.

yuvi-mittal commented:
@apoorvsxna , Can you please add more tests using for and ifStmt ?

xmnlab commented:
thanks for working on that @apoorvsxna !
thanks for the review @yuvi-mittal !


@apoorvsxna, in general your PR LGTM. as mentioned by @yuvi-mittal , it would be nice to expand the tests for int16.

maybe you could reuse some current tests for int32, and parametrize it for int32 and int16 with pytest.parametrize.

apoorvsxna commented:
Hey @xmnlab . I've made some changes. Expanded the tests for int32. Implemented type promotion and added tests for that as well. Could you take a look please? Thanks!

xmnlab commented on src/irx/builders/llvmliteir.py:
```diff
@@ -81,7 +84,7 @@ class LLVMLiteIRVisitor(BuilderVisitor):
     """LLVM-IR Translator."""
 
     # AllocaInst
-    named_values: dict[str, Any] = {}  # noqa: RUF012
```
although it is not too much pythonic, but there is no need to remove it.
it is useful as a documentation

apoorvsxna replied:
> although it is not too much pythonic, but there is no need to remove it. it is useful as a documentation

Hey I'm not quite sure what was meant here. But I've left that line in now. Let me know if it needs to be changed.

xmnlab commented on src/irx/builders/llvmliteir.py:
```diff
@@ -90,8 +93,10 @@ class LLVMLiteIRVisitor(BuilderVisitor):
-        self.result_stack: list[ir.Value | ir.Function] = []
+        # named_values as instance variable so it isn't shared across instances
+        self.named_values: dict[str, Any] = {}
+        self.function_protos = {}
```
any reason for removing the type annotation here?

xmnlab commented on src/irx/builders/llvmliteir.py:
```diff
@@ -90,8 +93,10 @@ class LLVMLiteIRVisitor(BuilderVisitor):
+        # named_values as instance variable so it isn't shared across instances
+        self.named_values: dict[str, Any] = {}
+        self.function_protos = {}
+        self.result_stack = []
```
any reason for removing the type annotation here?

apoorvsxna commented:
@xmnlab Fixed the review issues and updated the PR.

apoorvsxna commented:
Hey @xmnlab . The CI checks show a linter error that doesn't come up in my local setup. I wasn't sure how to debug this so I just made the required changes and created a new PR #53 . Please take a look and let me know if looks good now. Thanks!

xmnlab commented:
@apoorvsxna thanks for working on that.

could you rebase this branch on top of upstream main, please?

btw use our discord for getting attention to a PR or for keeping the conversation flowing!

I will check the linting issue after you rebase your pr. thanks!

yuvi-mittal commented:
Heyy @xmnlab , the changes made in this PR have been merged, @apoorvsxna opened another PR for adding int16 support maybe because of trouble he was facing in rebasing or any other issue. So , we can close this PR.

xmnlab commented:
Thanks for letting me know @yuvi-mittal 
I knew I had seen that before XD It seems I need vacations hehe

yuvi-mittal commented:
Hey @xmnlab , you definitely deserve some rest. I’ll take care of a few issues on IRx and review the PRs, hopefully that helps free up some of your time.
